### PR TITLE
NAS-119476 / 23.10 / Add basic second step in new pool creation wizard

### PR DESCRIPTION
**Pull request by bvasilenko**
On **Storage** page, when user clicks **Create Pool (New)** , there's our new wizard.

### CODE REVIEW

File: src/app/pages/storage/modules/pool-manager/components/pool-manager-wizard/steps/create-data-wizard-step/create-data-wizard-step.component.html [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  <ix-select
+    formControlName="type"
+    [label]="'Type' | translate"
+    [tooltip]="'Select a VDEV layout' | translate"
+    [options]="vdevLayoutOptions$"
+    [required]="true"
+  ></ix-select>
+  <div class="content-container">
+    <div class="automad-disk-selection-container">
+      <h3>{{'Automad Disk Selection' | translate}}</h3>
```
Comment (undsoft) [inline src/app/pages/storage/modules/pool-manager/components/pool-manager-wizard/steps/create-data-wizard-step/create-data-wizard-step.component.html:12]: Typo in `Automad` and class name

File: src/app/pages/storage/modules/pool-manager/components/pool-manager-wizard/steps/create-data-wizard-step/create-data-wizard-step.component.html [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+      <ix-select
+        formControlName="number"
+        [label]="'Number of VDEVs' | translate"
+        [tooltip]="'Number of VDEVs to create' | translate"
+        [options]="numberOptions$"
+        [required]="true"
+      ></ix-select>
+    </div>
+    <div class="advaced-options-container">
+      <h3>{{'Advaced Options' | translate}}</h3>
```
Comment (undsoft) [inline src/app/pages/storage/modules/pool-manager/components/pool-manager-wizard/steps/create-data-wizard-step/create-data-wizard-step.component.html:36]: Maybe install a spell-checking plugin? 😄 
`Advanced`.

**Review by undsoft — APPROVED** [review]
Looks reasonable for now.
You can also add Encryption dropdown, like we have in the old pool manager and show it in configuration preview.

File: src/app/pages/storage/modules/pool-manager/components/configuration-preview/configuration-preview.component.html [hunk 3]
```diff
@@ -0,0 +1,37 @@
+<mat-card class="card">
+  <mat-card-header>
+    <h3 mat-card-title>{{ 'Configuration preview' | translate }}</h3>
```
Comment (undsoft) [inline src/app/pages/storage/modules/pool-manager/components/configuration-preview/configuration-preview.component.html:3]: `preview` => `Preview`

File: src/app/pages/storage/modules/pool-manager/components/inventory/inventory.component.ts [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+    const selectedSize = this.formValue.data.size_and_type[0];
+
+    const filterdSizeDisksMap: SizeDisksMap = {
+      hdd: !isHdd ? this.sizeDisksMap.hdd : {},
+      ssd: isHdd ? this.sizeDisksMap.ssd : {},
+    };
+
+    Object.entries(isHdd ? this.sizeDisksMap.hdd : this.sizeDisksMap.ssd).forEach(([size, number]) => {
+      if (isHdd) {
+        filterdSizeDisksMap.hdd[size] = size === selectedSize ? number - this.formValue.data.number : number;
```
[the remark this answers is not in the document]
Comment (bvasilenko) [reply src/app/pages/storage/modules/pool-manager/components/inventory/inventory.component.ts]: I think we can keep the existing logic for calculating of the remaining disks, but when it becomes possible to select different disks in the pool, we could just add the logic of excluding the selected disks from the remaining unused.

**Pull request merged**